groups/utils.js

```diff
@@ -178,6 +178,31 @@ function setParamValueInURL(paramKey, paramValue) {
   );
 }
 
+function fullDateString(timestamp) {
+  const options = {
+    weekday: 'long',
+    year: 'numeric',
+    month: 'short',
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+    timeZoneName: 'short',
+    hour12: true,
+  };
+  return new Intl.DateTimeFormat('en-US', options).format(new Date(timestamp));
+}
+
+function shortDateString(timestamp) {
```

Comment:
Add validation to ensure the timestamp is valid to prevent invalid date 
move it from the usage to the util function